query-engine/metrics/src/registry.rs

```diff
@@ -68,12 +62,14 @@ impl MetricRegistry {
         }
     }
 
-    fn record(&self, metric: &MetricVisitor) {
-        match metric.metric_type {
-            MetricType::Counter => self.handle_counter(metric),
-            MetricType::Gauge => self.handle_gauge(metric),
-            MetricType::Histogram => self.handle_histogram(metric),
-            MetricType::Description => self.handle_description(metric),
+    pub(crate) fn record(&self, metric: &MetricVisitor) {
+        if self.is_accepted_metric(metric) {
```

Comment:
Is it ever the case that we pass non-accepted metrics to `record()`?

Reply:
In tests, yes, we have explicit tests for this. In real code — I think right now probably not but I am not sure sure. It would happen if any library we depend on other than mobc was instrumented with metrics. So even if there are no other metrics producers right now, it's still worth to have this filter to avoid accidentally leaking unwanted metrics if any dependency were to suddenly add them.